collect/collect.go

```diff
@@ -270,6 +271,7 @@ func (i *InMemCollector) checkAlloc(ctx context.Context) {
 	if maxAlloc == 0 || mem.Alloc < uint64(maxAlloc) {
 		return
 	}
+	i.Metrics.Increment("collector_cache_eviction")
```

Comment:
Is it technically true that if we make it here we'll definitely evict something? An extreme example would be if the cache is empty we wouldnt, but this would still increment. More realistic would be everything matching the `countinue` conditions in the loop.

Reply:
`checkAlloc` is only called when `peer_queue` is empty. That would mean some spans are ingested and stored in the cache. I don't think it's possible this can increment without items in the cache

Reply:
Is it possible that non of the items in the cache would be evicted?

Reply:
I don't believe that would be possible in `concentrated` mode